### PR TITLE
Replace gnulib with libgcrypt

### PR DESCRIPTION
**Pull request by clumens**

### CODE REVIEW

File: lib/common/digest.c [hunk 1]
```diff
@@ -252,27 +254,39 @@ pcmk__xa_filterable(const char *name)
...
-        crm_trace("Digest %s.", digest);
 
-    } else {
-        crm_err("Could not create digest");
+    dlen = gcry_md_get_algo_dlen(GCRY_MD_MD5);
+    raw_digest = calloc(dlen, sizeof(unsigned char));
+
+    gcry_md_hash_buffer(GCRY_MD_MD5, raw_digest, buffer, strlen(buffer));
+
+    digest = calloc(2 * dlen, sizeof(char));
```
Comment (clumens) [inline lib/common/digest.c:281]: I can't tell if I need to add a +1 here to cover the null or not.

**Comment by kgaillot** [comment]
I wouldn't want to introduce a new dependency at a minor-minor release. Hang onto this for the future though.

**Comment by clumens** [comment]
> I wouldn't want to introduce a new dependency at a minor-minor release. Hang onto this for the future though.

One interesting thing to note here... I don't know what pulls it in, but /sbin/crm_*, /sbin/corosync, and /sbin/pacemaker* are already linked against libgcrypt, at least on RHEL9.

**Comment by kgaillot** [comment]
> > I wouldn't want to introduce a new dependency at a minor-minor release. Hang onto this for the future though.
> 
> One interesting thing to note here... I don't know what pulls it in, but `/sbin/crm_*`, `/sbin/corosync`, and `/sbin/pacemaker*` are already linked against libgcrypt, at least on RHEL9.

Interesting, possibly gnutls

**Comment by kgaillot** [comment]
> > > I wouldn't want to introduce a new dependency at a minor-minor release. Hang onto this for the future though.
> > 
> > 
> > One interesting thing to note here... I don't know what pulls it in, but `/sbin/crm_*`, `/sbin/corosync`, and `/sbin/pacemaker*` are already linked against libgcrypt, at least on RHEL9.
> 
> Interesting, possibly gnutls

looks like libdbus-1 on rhel8

**Pull request closed without merging**